loginas/utils.py

```diff
@@ -0,0 +1,63 @@
+import logging
+
+from django.conf import settings as django_settings
+from django.contrib.auth.models import User
+from django.contrib.auth import load_backend, login, logout
+from django.contrib import messages
+from django.core.signing import TimestampSigner, SignatureExpired
+from datetime import timedelta
+
+from . import settings as la_settings
+
+signer = TimestampSigner()
+
+logger = logging.getLogger(__name__)
+
+
+def login_as(user, request, store_original_user=True):
+    """
+    Utility function for forcing a login as specific user -- be careful about
+    calling this carelessly :)
+    """
+
+    # Save the original user pk before it is replaced in the login method
+    original_user_pk = request.user.pk
+
+    # Find a suitable backend.
+    if not hasattr(user, 'backend'):
+        for backend in django_settings.AUTHENTICATION_BACKENDS:
+            if user == load_backend(backend).get_user(user.pk):
+                user.backend = backend
+                break
+
+    # Log the user in.
+    if hasattr(user, 'backend'):
+        login(request, user)
+
```

Comment:
Do you think you can put an `else: return` here? I'm a bit uncomfortable with setting the flag even if we didn't manage to get a backend.